tests/system/Database/Live/ForgeTest.php

```diff
@@ -825,6 +826,41 @@ public function testAddColumn()
         $this->assertSame('username', $fieldNames[1]);
     }
 
+    public function testAddColumnNull()
+    {
+        $this->forge->dropTable('forge_test_table', true);
+
+        $this->forge->addField([
+            'col1' => ['type' => 'VARCHAR', 'constraint' => 255],
+            'col2' => ['type' => 'VARCHAR', 'constraint' => 255, 'null' => true],
+            'col3' => ['type' => 'VARCHAR', 'constraint' => 255, 'null' => false],
+        ]);
+        $this->forge->createTable('forge_test_table');
+
+        $this->forge->addColumn('forge_test_table', [
+            'col4' => ['type' => 'VARCHAR', 'constraint' => 255],
+            'col5' => ['type' => 'VARCHAR', 'constraint' => 255, 'null' => true],
+            'col6' => ['type' => 'VARCHAR', 'constraint' => 255, 'null' => false],
+        ]);
+
+        $this->db->resetDataCache();
+
+        $col1 = $this->getMetaData('col1', 'forge_test_table');
+        $this->assertFalse($col1->nullable);
+        $col2 = $this->getMetaData('col2', 'forge_test_table');
+        $this->assertTrue($col2->nullable);
+        $col3 = $this->getMetaData('col3', 'forge_test_table');
+        $this->assertFalse($col3->nullable);
+        $col4 = $this->getMetaData('col4', 'forge_test_table');
+        $this->assertTrue($col4->nullable);
```

Comment:
Since `$col1` is `false` from a logical point of view, I think `$col4` should be `false` too. Thoughts?

Reply:
See #7235
I thought that, but if we change it, it is a breaking change.

Reply:
Yes, fair point but in this case, I would treat it as a bug. 
API should be predictable and having a different behavior for the "same thing" is something we should avoid.

Reply:
Okay, I'll be fine with documenting this and leaving it to be fixed in the next major version.

@codeigniter4/database-team any thoughts on this one?

Reply:
It is inconsistent to have different behavior when creating and adding, but I am a little doubtful that it is worth changing in v5.

However, if we decide to change it, I would like to include that in the documentation.

@codeigniter4/database-team Opinions?

Reply:
I am ambivalent.

Reply:
It seems nobody has strong opinion.